gtests/net/tcp/common/set_sysctls.py

```diff
@@ -1,24 +1,46 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
+#
+# Copyright 2012 Google Inc. All Rights Reserved.
+
+"""Sets sysctl values and writes a file that restores them.
+
+The arguments are of the form "<proc-file>=<val>" separated by spaces.
+The program first reads the current value of the proc-file and creates
+a shell script named "/tmp/sysctl_restore_${PPID}.sh" which restores the values
+when executed. It then sets the new values
+"""
+
+__author__ = ('brakmo@google.com (Lawrence Brakmo)')
 
-import commands
 import os
+import subprocess
 import sys
 
-pppid = int(os.popen("ps -p %d -oppid=" % os.getppid()).read().strip())
-filename = '/tmp/sysctl_restore_%d.sh' % pppid
+# Sometimes shell skips fork() as an optimization.
+# https://unix.stackexchange.com/questions/466496/
+# If your shell does so, define SHELL_SKIPS_FORK=1
+# before running packetdrill.
+if os.environ.get('SHELL_SKIPS_FORK'):
+  filename = '/tmp/sysctl_restore_%d.sh' % os.getppid()
+else:
+  pppid = int(os.popen("ps -p %d -oppid=" % os.getppid()).read().strip())
+  filename = '/tmp/sysctl_restore_%d.sh' % pppid
```

Comment:
To reduce code duplication can we use something like:

    if os.environ.get('SHELL_SKIPS_FORK'):
        id = os.getppid()
    else:
        id = int(os.popen("ps -p %d -oppid=" % 
                      os.getppid()).read().strip())
    filename = '/tmp/sysctl_restore_%d.sh' % id

Reply:
With the PACKETDRILL_TEST_ID proposal above, we can ignore this part of the review. :-)

Reply:
Ack.